frontend/containers/sdgs/component.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC, useState, useEffect } from 'react';
+
+import { useIntl } from 'react-intl';
+
+import cx from 'classnames';
+
+import { uniq } from 'lodash';
+import { noop } from 'lodash';
+
+import { SDGS_DATA } from './constants';
+import SDG from './sdg';
+import type { SDGsProps } from './types';
+
+export const SDGs: FC<SDGsProps> = ({
```

Comment:
> I'm sorry if that's a big change but I wouldn't actually include the form part of the component.

No worries, doesn't have to be a huge change. I see 3 ways to go:  

1. Drop the ability to select SDGs from the PR entirely, turning it into a SDG list rendering only component.
    _To save some time, because this is a component required through the app and is a blocker to many of the views. We technically only need to worry about the selection/form part once we reach the profiles, though I suspect that can be soon._  
    - In this case, would we keep the wrapper `SDGs` component or just use the inner `SDG` one?
        _Here I think the wrapper would still be useful, as it can standardise sizing, so we can keep the spacing/sizing/gaps consistent in an easier way._

2. Add now a separate form component, as per suggestions  
    _Highly against this, mostly because of the blocking nature of this component. Plus we may still encounter issues when using it in the future so it'd probably need tweaking._  

3. Keep it for now but open a JIRA task to address the non-ideal accessibility, referencing this PR (so that it doesn't get lost). 
    _Not sure about this one. On one hand, we'd have an accessibility issue. On the other hand, we do unblock quite a few views by having an useable component._    

What are your thoughts? I think it's worth a chat on Monday though, because these pages have quite a few components that'll face similar issues/questions, and it'd be nice to figure out how to address these types of components in the future. 

> For one, I don't believe a toggle button is the best representation for a SDG when it's used in a form. If I look at native form elements, I would probably use a checkbox or radio button instead. If you use a screen reader on React Aria's [toggle button example](https://react-spectrum.adobe.com/react-aria/useToggleButton.html), you will hear it announced as “Test, toggle button not pressed”. Instead, if you'd use the useCheckbox hook, it would be announced as “Test, checkbox not ticked”. Not only this, but any input needs a label. How would a user know the SDG list is linked to a question?

Makes perfect sense! Although I did test it with VoiceOver, the label situation didn't even occur to me.  

> Secondly, and this is the reason why I'd have a separate component for forms: the UI does not seem to have any select with multiple selection. Instead, we have lists of options such as the SDGs' or [others](https://www.figma.com/proto/KC49uMR9t1YARmFR0Sr6aN/HeCo---Work-in-progress?node-id=2%3A4314&viewport=396%2C48%2C0.13&scaling=min-zoom&starting-point-node-id=2%3A5968&show-proto-sidebar=1). Since we may have lists that are visually quite different, I would prefer working on a component that works for them all. For example, that component could simply be a wrapper that delegate the rendering to the component that renders the form:

Not so sure about this, but I don't think I understood it. 

Going by the example, I anticipate we'd face an issue we have a lot, which is keeping things consistent design-wise throughout many different views. We'd either end up re-declaring the same tailwind styles in different views (due to a lack of a wrapper component dedicated to rendering that particular list's visual styles, or we'd end up prop-drilling. 

But again, I don't think I understood what you have in mind. 

> Finally, we discussed about using [React Hook Form](https://react-hook-form.com/) for managing form state and validation, so we'd need to make sure all our inputs are compatible with the library's types.

Agreed. 👍